modules/components/src/DataTable/utils.js

```diff
@@ -20,18 +20,33 @@ export function normalizeColumns({
     ...columnTypes,
     ...customTypes,
   };
+
   const mappedColumns = columns
-    .map(column => ({
-      ...column,
-      show: typeof column.show === 'boolean' ? column.show : true,
-      Cell: column.Cell || types[column.type],
-      hasCustomType: isNil(column.hasCustomType)
-        ? !!(customTypes || {})[column.type]
-        : column.hasCustomType,
-      ...(!column.accessor && !column.id ? { id: column.field } : {}),
-      ...(customTypeConfigs[column.type] || {}),
-    }))
+    .map(column => {
+      const customColIndex = customColumns.findIndex(
+        cc => cc.content.field === column.field,
+      );
+
+      let customCol = {};
+      if (customColIndex > -1) {
+        customCol = customColumns[customColIndex].content;
+        customColumns.splice(customColIndex, 1);
```

Comment:
This mutation feels troubling to me, especially in this case as `customColumns` is then used outside of this map. It's better imo to compute a new filtered array outside the map for use later.

Reply:
So before this mapping create two arrays "custom columns to use overriding" and a "custom columns to add"?

Reply:
I think after makes more sense, a const that's the filtered `customColumns` to feed in here https://github.com/overture-stack/arranger/pull/478/files/aaf642fcd7a51a475ff507fd9a9484c96126e6de#diff-f0ae8de5b156ad4c703b2c127b034abcR50